### PR TITLE
Initial implemenetation of save and restore

### PR DESCRIPTION
**Pull request by Ro4052**
Requires a slight change to the calling context from the viewer to reference the right component.

### CODE REVIEW

File: packages/perspective-viewer/src/js/viewer.js [hunk 1]
```diff
@@ -484,7 +484,7 @@ class PerspectiveViewer extends ActionElement {
             }
         }
         if (this._plugin.save) {
-            obj.plugin_config = this._plugin.save();
+            obj.plugin_config = this._plugin.save.call(this);
```
Comment (andy-lee-eng) [inline packages/perspective-viewer/src/js/viewer.js:487]: It looks like it should already do this, based on the way the other methods are called.

Maybe it's worth submitting this file's change directly to Perspective as a separate PR?

File: packages/perspective-viewer-d3fc/src/js/plugin/template.js [hunk 2]
```diff
@@ -72,6 +72,15 @@ class D3FCChartElement extends HTMLElement {
         return this._container;
     }
 
+    getSettings() {
+        return this._settings;
```
Comment (andy-lee-eng) [inline packages/perspective-viewer-d3fc/src/js/plugin/template.js]: I wonder if we should make sure we *only* save the user settings - i.e. don't save or restore the config that we get anyway via the `draw` method (row and column pivots, filters etc).

Comment (Ro4052) [reply packages/perspective-viewer-d3fc/src/js/plugin/template.js]: Yes, I agree. This was just an initial look but I think we need to decide what actually needs to be saved.

File: packages/perspective-viewer/src/js/viewer.js (hunk 1 above)
Comment (Ro4052) [reply packages/perspective-viewer/src/js/viewer.js:487]: I'll take a look at this now.

**Review by andy-lee-eng — APPROVED** [review]
I think we can probably take this in its current form.

If the PR to change the way the save/restore calls is rejected, or the API changes, we can adjust it then.

**Pull request merged**